### PR TITLE
Return pull chart error type on when pulling failed

### PR DESCRIPTION
**Pull request by tomahawk28**
Following https://github.com/giantswarm/chart-operator/pull/308#discussion_r338917401

> Is there a more specific error that we can match against? My thinking is that there is a specific case in which we really want to cancel. On the other hand there are cases which we do not expect and we should treat them like real errors.

### CODE REVIEW

File: pull_chart_tarball.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (c *Client) doFile(ctx context.Context, req *http.Request) (string, error)
 			// Github pages 404 produces full HTML page which
 			// obscures the logs.
 			if resp.StatusCode == http.StatusNotFound {
-				return backoff.Permanent(microerror.Maskf(executionFailedError, fmt.Sprintf("got StatusCode %d for url %#q", resp.StatusCode, req.URL.String())))
+				return backoff.Permanent(microerror.Maskf(pullChartFailedError, fmt.Sprintf("got StatusCode %d for url %#q", resp.StatusCode, req.URL.String())))
```
Comment (xh3b4sd) [inline pull_chart_tarball.go]: Why do we want to treat these two cases the same way? Isn't there any value in differentiating them? "Not Found" is a pretty different root cause than "No Such Host". Why ruling them all the same way?

Comment (tomahawk28) [reply pull_chart_tarball.go]: There's no serious reason against differentiating it, the only reason was that I don't want to have multiple `if else` statement in `chart-operator` when matcher function in using.

Comment (tomahawk28) [reply pull_chart_tarball.go]: I would return `notFound` error in this case and return `pullChartURLError` for other cases, WDYT?

Comment (rossf7) [reply pull_chart_tarball.go]: I'm fine with `pullChartFailedError` for both.

We need to pull the chart tarball over HTTP and it doesn't work. That is what we care about. The reason why is clear from the error text.

**Comment by tomahawk28** [comment]
an Issue created for CI: https://github.com/giantswarm/giantswarm/issues/7390

**Pull request merged**